MPL 3.6 event handling has changed

[projectroot.qt.applications.workbench.workbench.test_figureinteraction.test_figureinteraction](https://builds.mantidproject.org/job/pull_requests-conda-linux/1829/testReport/junit/projectroot.qt.applications/workbench/workbench_test_figureinteraction_test_figureinteraction/)

```
======================================================================
FAIL: test_construction_registers_handler_for_button_press_event (test_figureinteraction.FigureInteractionTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/qt/applications/workbench/workbench/plotting/test/test_figureinteraction.py", line 90, in test_construction_registers_handler_for_button_press_event
    fig_manager.canvas.mpl_connect.assert_has_calls(expected_call)
  File "/jenkins_workdir/workspace/pull_requests-conda-linux/mambaforge/envs/mantid-developer/lib/python3.8/unittest/mock.py", line 950, in assert_has_calls
    raise AssertionError(
AssertionError: Calls not found.
Expected: [call('button_press_event', <bound method FigureInteraction.on_mouse_button_press of <workbench.plotting.figureinteraction.FigureInteraction object at 0x7f59d25bf4c0>>),
 call('button_release_event', <bound method FigureInteraction.on_mouse_button_release of <workbench.plotting.figureinteraction.FigureInteraction object at 0x7f59d25bf4c0>>),
 call('draw_event', <bound method FigureInteraction.draw_callback of <workbench.plotting.figureinteraction.FigureInteraction object at 0x7f59d25bf4c0>>),
 call('motion_notify_event', <bound method FigureInteraction.motion_event of <workbench.plotting.figureinteraction.FigureInteraction object at 0x7f59d25bf4c0>>),
 call('resize_event', <bound method FigureInteraction.mpl_redraw_annotations of <workbench.plotting.figureinteraction.FigureInteraction object at 0x7f59d25bf4c0>>),
 call('figure_leave_event', <bound method FigureInteraction.on_leave of <workbench.plotting.figureinteraction.FigureInteraction object at 0x7f59d25bf4c0>>),
 call('axis_leave_event', <bound method FigureInteraction.on_leave of <workbench.plotting.figureinteraction.FigureInteraction object at 0x7f59d25bf4c0>>),
 call('scroll_event', <bound method FigureInteraction.on_scroll of <workbench.plotting.figureinteraction.FigureInteraction object at 0x7f59d25bf4c0>>)]
Actual: [call('button_press_event', <bound method FigureInteraction.on_mouse_button_press of <workbench.plotting.figureinteraction.FigureInteraction object at 0x7f59d25bf4c0>>),
 call('button_release_event', <bound method FigureInteraction.on_mouse_button_release of <workbench.plotting.figureinteraction.FigureInteraction object at 0x7f59d25bf4c0>>),
 call('draw_event', <bound method FigureInteraction.draw_callback of <workbench.plotting.figureinteraction.FigureInteraction object at 0x7f59d25bf4c0>>),
 call('motion_notify_event', <bound method FigureInteraction.motion_event of <workbench.plotting.figureinteraction.FigureInteraction object at 0x7f59d25bf4c0>>),
 call('resize_event', <bound method FigureInteraction.mpl_redraw_annotations of <workbench.plotting.figureinteraction.FigureInteraction object at 0x7f59d25bf4c0>>),
 call('figure_leave_event', <bound method FigureInteraction.on_leave of <workbench.plotting.figureinteraction.FigureInteraction object at 0x7f59d25bf4c0>>),
 call('scroll_event', <bound method FigureInteraction.on_scroll of <workbench.plotting.figureinteraction.FigureInteraction object at 0x7f59d25bf4c0>>)]
```

Also I had to remove this line in qt/applications/workbench/workbench/plotting/figureinteraction.py as the axis_leave_event has been deprecated        `self._cids.append(canvas.mpl_connect('axis_leave_event', self.on_leave))`

--- a/Framework/PythonInterface/mantid/plots/legend.py
+++ b/Framework/PythonInterface/mantid/plots/legend.py
@@ -66,7 +66,7 @@ class LegendProperties(dict):
         else:
             props['round_edges'] = False
 
-        props['columns'] = legend._ncol
+        props['columns'] = legend._ncols
         props['column_spacing'] = legend.columnspacing
         props['label_spacing'] = legend.labelspacing
 
@@ -132,7 +132,7 @@ class LegendProperties(dict):
 
         if int(matplotlib.__version__[0]) >= 2:
             legend = ax.legend(handles=get_legend_handles(ax),
-                               ncol=props['columns'],
+                               ncols=props['columns'],
                                prop={'size': props['entries_size']},
                                numpoints=props['markers'],
                                markerfirst=props['marker_position'] == "Left of Entries",
@@ -151,7 +151,7 @@ class LegendProperties(dict):
                                loc=loc)
         else:
             legend = ax.legend(handles=get_legend_handles(ax),
-                               ncol=props['columns'],
+                               ncols=props['columns'],
                                prop={'size': props['entries_size']},
                                numpoints=props['markers'],
                                markerfirst=props['marker_position'] == "Left of Entries",

--- a/Framework/PythonInterface/mantid/plots/legend.py
+++ b/Framework/PythonInterface/mantid/plots/legend.py
@@ -10,6 +10,8 @@
 """
 Functionality for dealing with legends on plots
 """
+from distutils.version import LooseVersion
+
 import matplotlib
 from matplotlib.patches import BoxStyle
 
@@ -66,7 +68,10 @@ class LegendProperties(dict):
         else:
             props['round_edges'] = False
 
-        props['columns'] = legend._ncols
+        if LooseVersion(matplotlib._version__) >= LooseVersion('3.6.0'):
+            props['columns'] = legend._ncols
+        else:
+            props['columns'] = legend._ncol
         props['column_spacing'] = legend.columnspacing
         props['label_spacing'] = legend.labelspacing
 
@@ -130,7 +135,7 @@ class LegendProperties(dict):
         if 'loc' in props.keys():
             loc = props['loc']
 
-        if int(matplotlib.__version__[0]) >= 2:
+        if LooseVersion(matplotlib._version__) >= LooseVersion('3.6.0'):
             legend = ax.legend(handles=get_legend_handles(ax),
                                ncols=props['columns'],
                                prop={'size': props['entries_size']},
@@ -151,7 +156,7 @@ class LegendProperties(dict):
                                loc=loc)
         else:
             legend = ax.legend(handles=get_legend_handles(ax),
-                               ncols=props['columns'],
+                               ncol=props['columns'],
                                prop={'size': props['entries_size']},
                                numpoints=props['markers'],
                                markerfirst=props['marker_position'] == "Left of Entries",
@@ -159,6 +164,8 @@ class LegendProperties(dict):
                                fancybox=props['round_edges'],
                                shadow=props['shadow'],
                                framealpha=props['transparency'],
+                               facecolor=props['background_color'],
+                               edgecolor=props['edge_color'],
                                title=props['title'],
                                borderpad=props['border_padding'],
                                labelspacing=props['label_spacing'],

--- a/Framework/PythonInterface/mantid/plots/legend.py
+++ b/Framework/PythonInterface/mantid/plots/legend.py
@@ -68,7 +68,7 @@ class LegendProperties(dict):
         else:
             props['round_edges'] = False
 
-        if LooseVersion(matplotlib._version__) >= LooseVersion('3.6.0'):
+        if LooseVersion(matplotlib.__version__) >= LooseVersion('3.6.0'):
             props['columns'] = legend._ncols
         else:
             props['columns'] = legend._ncol
@@ -135,7 +135,7 @@ class LegendProperties(dict):
         if 'loc' in props.keys():
             loc = props['loc']
 
-        if LooseVersion(matplotlib._version__) >= LooseVersion('3.6.0'):
+        if LooseVersion(matplotlib.__version__) >= LooseVersion('3.6.0'):
             legend = ax.legend(handles=get_legend_handles(ax),
                                ncols=props['columns'],
                                prop={'size': props['entries_size']},

--- a/Framework/PythonInterface/mantid/plots/utility.py
+++ b/Framework/PythonInterface/mantid/plots/utility.py
@@ -14,7 +14,7 @@ from distutils.version import LooseVersion
 # 3rd party imports
 from matplotlib import colors
 from matplotlib.legend import Legend
-from matplotlib import cm, __version__ as mpl_version_str
+from matplotlib import colormaps, __version__ as mpl_version_str
 from matplotlib.container import ErrorbarContainer
 
 # -----------------------------------------------------------------------------
@@ -110,7 +110,7 @@ def find_errorbar_container(line, containers):
 
 
 def get_colormap_names():
-    return sorted([cmap for cmap in cm.cmap_d.keys() if not cmap.endswith('_r')])
+    return sorted([cmap for cmap in colormaps.keys() if not cmap.endswith('_r')])
 
 
 def get_errorbar_containers(ax):
@@ -253,7 +253,7 @@ def get_single_workspace_log_value(ws_index, *, log_values=None, matrix_ws=None,
 
 def colormap_as_plot_color(number_colors: int, colormap_name: str = 'viridis', cmap=None):
     if not cmap:
-        cmap = cm.get_cmap(name=colormap_name)
+        cmap = colormaps[colormap_name]
 
     for i in range(number_colors):
         yield cmap(float(i) / number_colors)

--- a/mantid-developer-linux.yml
+++ b/mantid-developer-linux.yml
@@ -19,7 +19,7 @@ dependencies:
   - jemalloc=5.2.0 # We have noticed 5.2.1 can cause a hang on older glibc versions (Cent OS 7, Ubuntu 18.04). Observed with import CaChannel and also our MeierTest.
   - jsoncpp>=1.9.4,<2
   - librdkafka>=1.6.0
-  - matplotlib=3.5.*
+  - matplotlib=3.6.*
   - muparser>=2.3.2
   - nexus=4.4.*
   - ninja>=1.10.2

--- a/mantid-developer-osx.yml
+++ b/mantid-developer-osx.yml
@@ -19,7 +19,7 @@ dependencies:
   - jsoncpp>=1.9.4,<2
   - librdkafka>=1.6.0
   - muparser>=2.3.2
-  - matplotlib=3.5.*
+  - matplotlib=3.6.*
   - nexus=4.4.*
   - ninja>=1.10.2
   - numpy>=1.20.2

--- a/mantid-developer-win.yml
+++ b/mantid-developer-win.yml
@@ -20,7 +20,7 @@ dependencies:
   - librdkafka>=1.6.0
   - lib3mf # windows only
   - muparser>=2.3.2
-  - matplotlib=3.5.*
+  - matplotlib=3.6.*
   - nexus=4.4.*
   - ninja>=1.10.2
   - numpy>=1.20.2

--- a/qt/applications/workbench/workbench/plotting/figureinteraction.py
+++ b/qt/applications/workbench/workbench/plotting/figureinteraction.py
@@ -89,7 +89,6 @@ class FigureInteraction(object):
         self._cids.append(canvas.mpl_connect('motion_notify_event', self.motion_event))
         self._cids.append(canvas.mpl_connect('resize_event', self.mpl_redraw_annotations))
         self._cids.append(canvas.mpl_connect('figure_leave_event', self.on_leave))
-        self._cids.append(canvas.mpl_connect('axis_leave_event', self.on_leave))
         self._cids.append(canvas.mpl_connect('scroll_event', self.on_scroll))
 
         self.canvas = canvas

--- a/qt/applications/workbench/workbench/plotting/test/test_figureinteraction.py
+++ b/qt/applications/workbench/workbench/plotting/test/test_figureinteraction.py
@@ -84,7 +84,6 @@ class FigureInteractionTest(unittest.TestCase):
             call('motion_notify_event', interactor.motion_event),
             call('resize_event', interactor.mpl_redraw_annotations),
             call('figure_leave_event', interactor.on_leave),
-            call('axis_leave_event', interactor.on_leave),
             call('scroll_event', interactor.on_scroll)
         ]
         fig_manager.canvas.mpl_connect.assert_has_calls(expected_call)

--- a/qt/applications/workbench/workbench/plotting/test/test_figurewindow.py
+++ b/qt/applications/workbench/workbench/plotting/test/test_figurewindow.py
@@ -49,9 +49,11 @@ class Test(unittest.TestCase):
         # Find the center of the axes and simulate a drop event there
         # Need to use Qt logical pixels to factor in dpi
         ax_x_centre = (ax.xaxis.clipbox.min[0] + ax.xaxis.clipbox.width*0.5)/dpi_ratio
-        ax_y_centre = (ax.xaxis.clipbox.min[1] + ax.xaxis.clipbox.height*0.5)/dpi_ratio
-        mock_event = Mock(pos=lambda: Mock(x=lambda: ax_x_centre, y=lambda: ax_y_centre))
+        ax_y_centre = (ax.yaxis.clipbox.min[1] + ax.yaxis.clipbox.height*0.5)/dpi_ratio
+        mock_pos = Mock(position=lambda: Mock(x=lambda: ax_x_centre, y=lambda: ax_y_centre))
+        mock_event = Mock()
         mock_event.mimeData().text.return_value = "ws"
+        mock_event.pos.return_value = mock_pos
         with patch('workbench.plotting.figurewindow.QMainWindow.dropEvent'):
             self.fig_window.dropEvent(mock_event)
 

--- a/qt/python/mantidqt/mantidqt/plotting/test/test_functions.py
+++ b/qt/python/mantidqt/mantidqt/plotting/test/test_functions.py
@@ -12,7 +12,7 @@ from unittest import TestCase, main, mock
 
 # third party imports
 import matplotlib
-from matplotlib import cm
+from matplotlib import colormaps
 
 matplotlib.use('AGG')  # noqa
 import matplotlib.pyplot as plt
@@ -206,7 +206,7 @@ class FunctionsTest(TestCase):
     @mock.patch('mantidqt.plotting.functions.ConfigService', new_callable=MockConfigService)
     def test_pcolor_mesh_from_names_gets_colormap_from_ConfigService(self, mock_ConfigService):
         ws = CreateSampleWorkspace()
-        spring_colormap = cm.get_cmap('spring')
+        spring_colormap = colormaps['spring']
 
         fig = pcolormesh_from_names([ws])
 

--- a/qt/python/mantidqt/mantidqt/project/plotsloader.py
+++ b/qt/python/mantidqt/mantidqt/project/plotsloader.py
@@ -9,9 +9,8 @@
 import copy
 
 import matplotlib.axes
-import matplotlib.cm as cm
 import matplotlib.colors
-from matplotlib import axis, ticker  # noqa
+from matplotlib import axis, ticker, colormaps  # noqa
 from matplotlib.ticker import NullFormatter,\
     ScalarFormatter, LogFormatterSciNotation
 
@@ -402,11 +401,11 @@ class PlotsLoader(object):
         # colorbar = image.colorbar
         image.set_clim(*sorted([dic["min"], dic["max"]]))
         image.set_label(dic["label"])
-        image.set_cmap(cm.get_cmap(dic["cmap"]))
+        image.set_cmap(colormaps[dic["cmap"]])
         image.set_interpolation(dic["interpolation"])
         # Try and make the cmap line up but sometimes it wont
         try:
-            image.axes.set_cmap(cm.get_cmap(dic["cmap"]))
+            image.axes.set_cmap(colormaps[dic["cmap"]])
         except AttributeError as e:
             logger.debug(
                 "PlotsLoader - The Image accessed did not have an axes with the ability to set the cmap: "

--- a/qt/python/mantidqt/mantidqt/widgets/codeeditor/completion.py
+++ b/qt/python/mantidqt/mantidqt/widgets/codeeditor/completion.py
@@ -49,7 +49,7 @@ ArgSpec = namedtuple("ArgSpec", "args varargs keywords defaults")
 def _ignore_matplotlib_deprecation_warnings():
     """Context-manager to disable deprecation warnings from matplotlib while
     generating the call tips"""
-    from matplotlib.cbook import MatplotlibDeprecationWarning
+    from matplotlib import MatplotlibDeprecationWarning
     with warnings.catch_warnings():
         warnings.simplefilter("ignore", MatplotlibDeprecationWarning)
         yield

--- a/qt/python/mantidqt/mantidqt/widgets/colorbar/colorbar.py
+++ b/qt/python/mantidqt/mantidqt/widgets/colorbar/colorbar.py
@@ -17,7 +17,7 @@ from matplotlib.colorbar import Colorbar
 from matplotlib.figure import Figure
 from matplotlib.colors import ListedColormap, Normalize, SymLogNorm, PowerNorm, LogNorm
 from matplotlib.image import AxesImage
-from matplotlib import cm
+from matplotlib import colormaps
 import numpy as np
 from qtpy.QtWidgets import QWidget, QVBoxLayout, QHBoxLayout, QLineEdit, QComboBox, QCheckBox, QLabel
 from qtpy.QtCore import Signal, Qt
@@ -35,10 +35,10 @@ def register_customized_colormaps():
         cmap_name = os.path.basename(cmap_file).split(".")[0]
         cmap_data = np.loadtxt(cmap_file)/255.0
         cmap = ListedColormap(cmap_data, name=cmap_name)
-        cm.register_cmap(name=cmap_name, cmap=cmap)
+        colormaps.register(name=cmap_name, cmap=cmap)
         cmap_data_r = np.flipud(cmap_data)
         cmap_r = ListedColormap(cmap_data_r, name=f"{cmap_name}_r")
-        cm.register_cmap(name=f"{cmap_name}_r", cmap=cmap_r)
+        colormaps.register(name=f"{cmap_name}_r", cmap=cmap_r)
 
 
 class ColorbarWidget(QWidget):
@@ -47,7 +47,7 @@ class ColorbarWidget(QWidget):
     # register additional color maps from file
     register_customized_colormaps()
     # create the list
-    cmap_list = sorted([cmap for cmap in cm.cmap_d.keys() if not cmap.endswith('_r')])
+    cmap_list = sorted([cmap for cmap in colormaps.keys() if not cmap.endswith('_r')])
 
     def __init__(self, parent=None, default_norm_scale=None):
         """
@@ -193,7 +193,7 @@ class ColorbarWidget(QWidget):
         self.cmin_value = low
         self.cmax_value = high
         self.update_clim_text()
-        self.cmap.setCurrentIndex(sorted(cm.cmap_d.keys()).index(mappable_cmap.name))
+        self.cmap.setCurrentIndex(sorted(colormaps.keys()).index(mappable_cmap.name))
         self.redraw()
 
     def norm_changed(self):

--- a/qt/python/mantidqt/mantidqt/widgets/colorbar/colorbar.py
+++ b/qt/python/mantidqt/mantidqt/widgets/colorbar/colorbar.py
@@ -285,7 +285,7 @@ class ColorbarWidget(QWidget):
         Redraws the colobar and emits signal to cause the parent to redraw
         """
         self.colorbar.update_ticks()
-        self.colorbar.draw_all()
+        self.canvas.figure.draw_without_rendering()
         self.canvas.draw_idle()
         self.colorbarChanged.emit()
 

--- a/qt/python/mantidqt/mantidqt/widgets/plotconfigdialog/imagestabwidget/view.py
+++ b/qt/python/mantidqt/mantidqt/widgets/plotconfigdialog/imagestabwidget/view.py
@@ -8,7 +8,7 @@
 
 import numpy as np
 from matplotlib.colors import LogNorm, Normalize
-from matplotlib import cm
+from matplotlib import colormaps
 from qtpy.QtCore import Qt
 from qtpy.QtGui import QPixmap, QIcon, QImage
 from qtpy.QtWidgets import QWidget
@@ -26,7 +26,7 @@ SCALES = {'Linear': Normalize, 'Logarithmic': LogNorm}
 
 
 def create_colormap_img(cmap_name, width=50, height=20):
-    colormap = cm.get_cmap(cmap_name)
+    colormap = colormaps[cmap_name]
     gradient_array = np.tile(np.linspace(0, 1, width), height)
     img_array = (colormap(gradient_array)*255).astype(np.uint8)
     return QImage(img_array, width, height, QImage.Format_RGBA8888_Premultiplied)

--- a/qt/python/mantidqt/mantidqt/widgets/plotconfigdialog/test/test_apply_all_properties.py
+++ b/qt/python/mantidqt/mantidqt/widgets/plotconfigdialog/test/test_apply_all_properties.py
@@ -389,7 +389,10 @@ class ApplyAllPropertiesTest(unittest.TestCase):
                          isinstance(self.new_legend.legendPatch.get_boxstyle(), BoxStyle.Round))
 
     def test_apply_properties_on_figure_with_legend_sets_number_of_columns(self):
-        self.assertEqual(new_legend_props['columns'], self.new_legend._ncol)
+        if LooseVersion(matplotlib.__version__) >= LooseVersion('3.6.0'):
+            self.assertEqual(new_legend_props['columns'], self.new_legend._ncols)
+        else:
+            self.assertEqual(new_legend_props['columns'], self.new_legend._ncol)
 
     def test_apply_properties_on_figure_with_legend_sets_column_spacing(self):
         self.assertEqual(new_legend_props['column_spacing'], self.new_legend.columnspacing)

--- a/qt/python/mantidqt/mantidqt/widgets/regionselector/test/test_regionselector_presenter.py
+++ b/qt/python/mantidqt/mantidqt/widgets/regionselector/test/test_regionselector_presenter.py
@@ -6,7 +6,7 @@
 # SPDX - License - Identifier: GPL - 3.0 +
 import unittest
 
-from unittest.mock import call, DEFAULT, Mock, patch
+from unittest.mock import call, DEFAULT, Mock, patch, MagicMock
 
 from mantidqt.widgets.regionselector.presenter import RegionSelector
 from mantidqt.widgets.sliceviewer.models.workspaceinfo import WS_TYPE
@@ -19,6 +19,8 @@ class RegionSelectorTest(unittest.TestCase):
                                                WorkspaceInfo=DEFAULT)
         self.patched_deps = self._ws_info_patcher.start()
         self.patched_deps["WorkspaceInfo"].get_ws_type.return_value = WS_TYPE.MATRIX
+        self.mock_view = MagicMock()
+        self.mock_view._data_view.ax._get_aspect_ratio.return_value = 1
 
     def tearDown(self) -> None:
         self._ws_info_patcher.stop()
@@ -68,7 +70,7 @@ class RegionSelectorTest(unittest.TestCase):
         mock_view.set_workspace.assert_called_once_with(mock_ws)
 
     def test_add_rectangular_region_creates_selector(self):
-        region_selector = RegionSelector(ws=Mock(), view=Mock())
+        region_selector = RegionSelector(ws=Mock(), view=self.mock_view)
 
         region_selector.add_rectangular_region("test", "black")
 
@@ -77,7 +79,7 @@ class RegionSelectorTest(unittest.TestCase):
         self.assertEqual("test", region_selector._selectors[0].region_type())
 
     def test_add_second_rectangular_region_deactivates_first_selector(self):
-        region_selector = RegionSelector(ws=Mock(), view=Mock())
+        region_selector = RegionSelector(ws=Mock(), view=self.mock_view)
 
         region_selector.add_rectangular_region("test", "black")
         region_selector._drawing_region = False
@@ -89,7 +91,7 @@ class RegionSelectorTest(unittest.TestCase):
         self.assertTrue(region_selector._selectors[1].active)
 
     def test_clear_workspace_will_clear_all_the_selectors_and_model_workspace(self):
-        region_selector = RegionSelector(view=Mock())
+        region_selector = RegionSelector(view=self.mock_view)
         mock_ws = Mock()
 
         region_selector.update_workspace(mock_ws)
@@ -266,7 +268,7 @@ class RegionSelectorTest(unittest.TestCase):
         region_selector.view.set_override_cursor.assert_called_once_with(True)
 
     def test_on_rectangle_selected_notifies_observer(self):
-        region_selector = RegionSelector(ws=Mock(), view=Mock())
+        region_selector = RegionSelector(ws=Mock(), view=self.mock_view)
         mock_observer = Mock()
         region_selector.subscribe(mock_observer)
 
@@ -276,7 +278,7 @@ class RegionSelectorTest(unittest.TestCase):
         mock_observer.notifyRegionChanged.assert_called_once()
 
     def test_cancel_drawing_region_will_remove_last_selector(self):
-        region_selector = RegionSelector(ws=Mock(), view=Mock())
+        region_selector = RegionSelector(ws=Mock(), view=self.mock_view)
         region_selector.add_rectangular_region("test", "black")
         self.assertEqual(1, len(region_selector._selectors))
         region_selector.cancel_drawing_region()
@@ -284,7 +286,7 @@ class RegionSelectorTest(unittest.TestCase):
 
     def test_when_multiple_region_adds_are_requested_only_one_region_is_added(self):
         # Given
-        region_selector = RegionSelector(ws=Mock(), view=Mock())
+        region_selector = RegionSelector(ws=Mock(), view=self.mock_view)
         region_selector.add_rectangular_region("test", "black")
         self.assertEqual(1, len(region_selector._selectors))
 

--- a/qt/python/mantidqtinterfaces/mantidqtinterfaces/simplescanviewer/rectangle_selection.py
+++ b/qt/python/mantidqtinterfaces/mantidqtinterfaces/simplescanviewer/rectangle_selection.py
@@ -5,8 +5,6 @@
 #   Institut Laue - Langevin & CSNS, Institute of High Energy Physics, CAS
 # SPDX - License - Identifier: GPL - 3.0 +
 
-import warnings
-
 from matplotlib.widgets import RectangleSelector
 
 
@@ -25,55 +23,46 @@ class RectangleSelection(RectangleSelector):
         So for now, we use this version even if there is no rational reason I can see for it to be needed.
         """
 
-        # matplotlib is bound to deprecate the eventpress attribute with the release of version 3.7, making it private
-        # In the meantime, it gives off warnings every time it is accessed.
-        # But since this feature does not work correctly at the moment, and we need this fix as a workaround, we just
-        # silence the warnings and hope that this bug will be understood and solved by the time this deprecation
-        # actually comes into effect.
-        # This is not nearly a perennial fix, but it should hold for a year or so.
+        if not self._interactive:
+            self.to_draw.set_visible(False)
 
-        with warnings.catch_warnings():
-            warnings.filterwarnings("ignore", category=DeprecationWarning)
-            if not self.interactive:
-                self.to_draw.set_visible(False)
+        # update the eventpress and eventrelease with the resulting extents
+        x1, x2, y1, y2 = self.extents
+        self._eventpress.xdata = x1
+        self._eventpress.ydata = y1
+        xy1 = self.ax.transData.transform([x1, y1])
+        self._eventpress.x, self._eventpress.y = xy1
 
-            # update the eventpress and eventrelease with the resulting extents
-            x1, x2, y1, y2 = self.extents
-            self.eventpress.xdata = x1
-            self.eventpress.ydata = y1
-            xy1 = self.ax.transData.transform([x1, y1])
-            self.eventpress.x, self.eventpress.y = xy1
+        self._eventrelease.xdata = x2
+        self._eventrelease.ydata = y2
+        xy2 = self.ax.transData.transform([x2, y2])
+        self._eventrelease.x, self._eventrelease.y = xy2
 
-            self.eventrelease.xdata = x2
-            self.eventrelease.ydata = y2
-            xy2 = self.ax.transData.transform([x2, y2])
-            self.eventrelease.x, self.eventrelease.y = xy2
+        # calculate dimensions of box or line
+        if self.spancoords == 'data':
+            spanx = abs(self._eventpress.xdata - self._eventrelease.xdata)
+            spany = abs(self._eventpress.ydata - self._eventrelease.ydata)
+        elif self.spancoords == 'pixels':
+            spanx = abs(self._eventpress.x - self._eventrelease.x)
+            spany = abs(self._eventpress.y - self._eventrelease.y)
+        else:
+            raise ValueError('spancoords must be "data" or "pixels"')
 
-            # calculate dimensions of box or line
-            if self.spancoords == 'data':
-                spanx = abs(self.eventpress.xdata - self.eventrelease.xdata)
-                spany = abs(self.eventpress.ydata - self.eventrelease.ydata)
-            elif self.spancoords == 'pixels':
-                spanx = abs(self.eventpress.x - self.eventrelease.x)
-                spany = abs(self.eventpress.y - self.eventrelease.y)
-            else:
-                raise ValueError('spancoords must be "data" or "pixels"')
-
-            # check if drawn distance (if it exists) is not too small in
-            # either x or y-direction
-            if (self.drawtype != 'none'
-                    and (self.minspanx is not None and spanx < self.minspanx
-                         or self.minspany is not None and spany < self.minspany)):
-                for artist in self.artists:
-                    artist.set_visible(False)
-                self.update()
-                return
-
-            # call desired function
-            self.onselect(self.eventpress, self.eventrelease)
+        # check if drawn distance (if it exists) is not too small in
+        # either x or y-direction
+        if (self._drawtype != 'none'
+                and (self.minspanx is not None and spanx < self.minspanx
+                     or self.minspany is not None and spany < self.minspany)):
+            for artist in self.artists:
+                artist.set_visible(False)
             self.update()
+            return
 
-            return False
+        # call desired function
+        self.onselect(self._eventpress, self._eventrelease)
+        self.update()
+
+        return False
 
     def onmove(self, event):
         """

--- a/qt/python/mantidqtinterfaces/test/SimpleScanViewer/RectanglePlotTest.py
+++ b/qt/python/mantidqtinterfaces/test/SimpleScanViewer/RectanglePlotTest.py
@@ -31,6 +31,7 @@ class RectanglePlotTest(unittest.TestCase):
         config['default.instrument'] = "D16"
 
         plotter = mock.MagicMock()
+        plotter.image_axes._get_aspect_ratio.return_value = 1
         exporter = mock.MagicMock()
         exporter.rectangles_manager = RectanglesManager()
 

--- a/qt/widgets/mplcpp/test/FigureCanvasQtTest.h
+++ b/qt/widgets/mplcpp/test/FigureCanvasQtTest.h
@@ -22,20 +22,24 @@ public:
 public:
   void testConstructionYieldsExpectedSubplot() {
     FigureCanvasQt canvas{111};
-    auto geometry = canvas.gca().pyobj().attr("get_geometry")();
+    auto subplotspec = canvas.gca().pyobj().attr("get_subplotspec")();
+    auto geometry = subplotspec.attr("get_geometry")();
     TS_ASSERT_EQUALS(1, geometry[0]);
     TS_ASSERT_EQUALS(1, geometry[1]);
-    TS_ASSERT_EQUALS(1, geometry[2]);
+    TS_ASSERT_EQUALS(0, geometry[2]);
+    TS_ASSERT_EQUALS(0, geometry[3]);
   }
 
   void testConstructionCapturesGivenAxesObject() {
     Figure fig;
     fig.addSubPlot(221);
     FigureCanvasQt canvas{std::move(fig)};
-    auto geometry = canvas.gca().pyobj().attr("get_geometry")();
+    auto subplotspec = canvas.gca().pyobj().attr("get_subplotspec")();
+    auto geometry = subplotspec.attr("get_geometry")();
     TS_ASSERT_EQUALS(2, geometry[0]);
     TS_ASSERT_EQUALS(2, geometry[1]);
-    TS_ASSERT_EQUALS(1, geometry[2]);
+    TS_ASSERT_EQUALS(0, geometry[2]);
+    TS_ASSERT_EQUALS(0, geometry[3]);
   }
 
   void testToDataCoordinatesReturnsExpectedPoint() {

--- a/scripts/ExternalInterfaces/CMakeLists.txt
+++ b/scripts/ExternalInterfaces/CMakeLists.txt
@@ -7,7 +7,7 @@ externalproject_add(
   mslice
   PREFIX ${_mslice_external_root}
   GIT_REPOSITORY "https://github.com/mantidproject/mslice.git"
-  GIT_TAG 425465fd4f7628b89b63596fe67f3e4bb7dba640
+  GIT_TAG f515a0ee17938ea90d77d6487e9bb4e4d4f95a74
   EXCLUDE_FROM_ALL 1
   CONFIGURE_COMMAND ""
   BUILD_COMMAND ""

